### ISSUE TITLE
Rework the interactive DST lesson

### ISSUE THREAD

**Issue by goi42** [lhcb/starterkit-lessons#57]
As written, with the copying and pasting of functions and restarting the ipython session, it's very clunky and hard-to-follow in class. It might be good to rework the lesson actually using Bender, as this is how people access the TES in real life.

**Comment by alexpearce** [comment]
We have discussed using Bender before. The reason for not using it at first is to show people how quick and easy it can be to have a poke around a file. There's a lot of magic in Bender and I prefer to minimise that when teaching.

I do think we can and should simplify those explorer functions, though.

**Comment by saschastahl** [comment]
Bender knowledge is also not directly transferable to other LHCb applications. E.g. when you develop things in Moore or Brunel, it is sometimes handy to write small GaudiPython snippets exploring the TES.

**Comment by goi42** [comment]
Hm these are good points. I think, though, if this lesson is really to be useful, there should be some detail on what those scripts are really doing included in the lesson. The appMgr, for instance, is never really explained, making the lesson as-is somewhat magical anyway.

Such explanations, though, will begin to eat up some time in class...

**Comment by alexpearce** [comment]
You're right. I think it is worth taking the time to explain every aspect of the explorer script, though. The ideas are fundamental for how we manipulate the data (either directly when exploring, or indirectly with a `DecayTreeTuple`).

See the teachers in this Starterkit is making me wonder if we need to expand the lessons to dive a little deeper into the basics (which would mean dropping some things).

**Comment by goi42** [comment]
I agree. Some lessons, like the data flow, are inevitably going to be an information dump but remain essential for providing context, but there is room for improvement in the pacing and detail of other lessons. I was glad, for instance, that Albert went over some of the TES stuff again in a more hands-on way during the LoKi session as it was clear the students hadn't really taken it in after my initial DST session, in which there wasn't time to really play with it.

Regarding dropping material, we can have some disconnect between the website and the actual lessons. That is, we can have more expansive lessons online for reference than we go over in the sessions. This would require some extra effort but might be worth it.

Unfortunately, I do not think I will be able to update this lesson myself as I don't work with the `appMgr` in my analysis; my experience with it comprises only my prep for teaching the session.
